pkg/controller/vitessshard/reconcile_disk.go

```diff
@@ -0,0 +1,129 @@
+package vitessshard
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/pkg/operator/results"
+	"planetscale.dev/vitess-operator/pkg/operator/rollout"
+)
+
+const (
+	pvcDiskSizeAnnotation = "planetscale.com/pvc-filesystem-resize"
+)
+
+func (r *ReconcileVitessShard) reconcileDisk(ctx context.Context, vts *planetscalev2.VitessShard) (reconcile.Result, error) {
+	resultBuilder := &results.Builder{}
+	anythingChanged := false
+	tabletPods, err := r.tabletPodsFromShard(ctx, vts)
+	if err != nil {
+		return resultBuilder.Error(err)
+	}
+
+	for i  := range vts.Spec.TabletPools {
+		tabletPool := &vts.Spec.TabletPools[i]
+		requestedDiskQuantity := tabletPool.DataVolumeClaimTemplate.Resources.Requests[v1.ResourceStorage]
+		requestedDisk := requestedDiskQuantity.String()
+		poolTablets, err := tabletsForPool(vts, tabletPool.Cell, string(tabletPool.Type))
+		if err != nil {
+			return resultBuilder.Error(err)
+		}
+
+		for _, tabletKey := range poolTablets {
+			pod, ok := tabletPods[tabletKey]
+			if !ok {
+				continue
+			}
+
+			pvc, err := r.claimForTabletPod(ctx, pod)
+			if err != nil {
+				return resultBuilder.Error(err)
+			}
+
+			// If the PVC does not have the FileSystemResizeCondition, bail out.
+			if !checkPVCFileSystemResizeCondition(pvc) {
+				return resultBuilder.Result()
```

Comment:
We don't know yet that there is a resize pending on this PVC. If we bail out for any PVC without the resize condition, we will be blocked if any PVCs are simply not being resized. We need to first skip (`continue`) any PVC whose Status size is already the same as the desired size.